kimchi/src/lagrange_basis_evaluations.rs

```diff
@@ -106,19 +173,35 @@ impl<F: FftField> LagrangeBasisEvaluations<F> {
 
         batch_inversion_and_mul(&mut denominators[..], &numerator);
 
-        // Denominators now contains the desired result.
+        // Denominators now contain the desired result.
         LagrangeBasisEvaluations {
             evals: vec![denominators],
         }
     }
 
-    /// Compute all evaluations of the normalized lagrange basis polynomials of the
-    /// given domain at the given point. Runs in time O(n log(n)) for n = domain size.
+    /// Compute all evaluations of the normalized Lagrange basis polynomials of
+    /// the given domain at the given point `x`. Runs in time O(n log(n)) where
+    /// n is the domain size.
     fn new_with_chunked_segments(
         max_poly_size: usize,
         domain: D<F>,
         x: F,
     ) -> LagrangeBasisEvaluations<F> {
+        // For each chunk, this loop obtains the coefficient form of the
+        // polynomial that equals the powers of `x` in the positions
+        // corresponding to the chunk, and 0 elsewhere in the domain, using an
+        // iFFT operation of length n, resulting in an algorithm that runs in
+        // O(c n log n).
+        //
+        // Example:
+        //                                  i-th chunk
+        //                          -----------------------
+        //   chunked: [ 0, ..., 0,  1, x, x^2, ..., x^{m-1}, 0, ..., 0 ]
+        //   indices:   0    i·m-1  i·m            (i+1)m-1  (i+1)m  cm-1=n-1
+        //
+        // A total of n coefficients are returned. These will be helpful to
+        // evaluate the chunks of polynomials of degree c·n at the point x.
+        //
         let n = domain.size();
```

Comment:
Note that everywhere in the code, there is no check about if n < max_poly_size (therefore chunks must be equal to 1, and not 0 as it is), and also there is no check when they do not divide each other (might a be a bug you're looking for). I would always add a check when we divide.

Reply:
And of course, add a test to check if it handled correctly.

Reply:
Actually the check that n < max_poly_size is performed here -> https://github.com/o1-labs/proof-systems/blob/fbf0572a456b7a6ced34dd493f62562d220d1274/kimchi/src/lagrange_basis_evaluations.rs#L243